homeassistant/components/home_connect/light.py

```diff
@@ -84,14 +85,58 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Home Connect light."""
+    known_appliance_ids: set[str] = set()
 
-    async_add_entities(
-        [
+    def get_entities_for_appliance(
+        appliance: HomeConnectApplianceData,
+    ) -> list[HomeConnectLight]:
+        """Get a list of entities."""
+        known_appliance_ids.add(appliance.info.ha_id)
+        return [
             HomeConnectLight(entry.runtime_data, appliance, description)
             for description in LIGHTS
-            for appliance in entry.runtime_data.data.values()
             if description.key in appliance.settings
-        ],
+        ]
+
+    async_add_entities(
+        entity
+        for appliance in entry.runtime_data.data.values()
+        for entity in get_entities_for_appliance(appliance)
+    )
+
+    def handle_paired_or_connected_appliance() -> None:
+        """Handle new paired appliance or a appliance that has been connected."""
+        for appliance in entry.runtime_data.data.values():
+            if appliance.info.ha_id in known_appliance_ids:
+                entity_registry = er.async_get(hass)
+                async_add_entities(
+                    entity
+                    for entity in get_entities_for_appliance(appliance)
+                    if not entity_registry.async_get(cast(str, entity.unique_id))
```

Comment:
`entity_registry.async_get` takes an `Entity.entity_id` or `entity_registry.RegistryEntry.id` argument not an `Entity.unique_id`.